frontend/src/API/use/useFeed.ts

```diff
@@ -44,7 +44,7 @@ export function useFeed(id: string, feedType: FeedType, page: number, perpage: n
                 })
```

Comment:
Bug: when switching between feed types (in this case from  `watch-all` to `subsite`)  `useEffect` in `useFeed` is triggered twice:
<img width="1413" alt="image" src="https://user-images.githubusercontent.com/2865203/170632329-0d758515-88e4-4f48-a985-4f9a38168012.png">As a result, current posts could be set to the value that corresponds to the previous feed.

I'm not sure why this happens, it seems that `FeedPage` is called in some transition state, when `AppState.site` is still "main". Perhaps this is an artifact of having `useSearchParams` and `useMatch` inside the `FeedPage`.

Reply:
Ok, here is what is happening:
When location is changed, in the rendering phase the the body of the `FeedPage` is rendered (and `feedType` is calculated). After that, during the mounting phase the effect hooks in `SiteWatcher` and `useEffect` are fired. The problem is, even though `site` is already changed, when `useFeed` effect is fired, the `feedType` has it's previous value, before site was changed.

A quick workaround would be to move `useMatch` into the same `SiteWatcher` component and make them part of the `AppState`.

I'll try to think about a better approach.

Reply:
I found a better way to fix this issue:
https://github.com/spaceshelter/orbitar/commit/11a044609380218ec23d3806a578c76338bcbc6d

I used [a small library](https://github.com/superwf/mobx-react-router) that bridges react-router and mobx, removing intermediate steps with hooks and effects from synchronization of mobx state with actual location and history.

Note, how `site` and `siteInfo` are now directly derived from the underlying state.

Reply:
Fixed in ad82828192517f0701ef3dc9b7a62e4a0cc0f3c6